app/models/collection.rb

```diff
@@ -52,170 +53,163 @@ def clean_up_challenge
   def collection_depth
-    if (self.parent && self.parent.parent) || (self.parent && !self.children.empty?) || (!self.children.empty? && !self.children.collect(&:children).flatten.empty?)
-      errors.add(:base, ts("Sorry, but %{name} is a subcollection, so it can't also be a parent collection.", name: parent.name))
-    end
+    errors.add(:base, I18n.t("collection.validation.collection_depth", name: parent.name)) if self.parent&.parent || (self.parent && !self.children.empty?) || (!self.children.empty? && !self.children.collect(&:children).flatten.empty?)
   end
 
   validate :parent_exists
   def parent_exists
-    unless parent_name.blank? || Collection.find_by(name: parent_name)
-      errors.add(:base, ts("We couldn't find a collection with name %{name}.", name: parent_name))
-    end
+    errors.add(:base, I18n.t("collection.validation.collection_not_found", name: parent_name)) unless parent_name.blank? || Collection.find_by(name: parent_name)
   end
 
   validate :parent_is_allowed
   def parent_is_allowed
-    if parent
-      if parent == self
-        errors.add(:base, ts("You can't make a collection its own parent."))
-      elsif parent_id_changed? && !parent.user_is_maintainer?(User.current_user)
-        errors.add(:base, ts("You have to be a maintainer of %{name} to make a subcollection.", name: parent.name))
-      end
+    return unless parent
+    
+    if parent == self
+      errors.add(:base, I18n.t("collection.validation.invalid_parent"))
+    elsif parent_id_changed? && !parent.user_is_maintainer?(User.current_user)
+      errors.add(:base, I18n.t("collection.validation.subcollection", name: parent.name))
     end
   end
 
-  validates_presence_of :name, message: ts("Please enter a name for your collection.")
-  validates :name, uniqueness: { message: ts("Sorry, that name is already taken. Try again, please!") }
-  validates_length_of :name,
-    minimum: ArchiveConfig.TITLE_MIN,
-    too_short: ts("must be at least %{min} characters long.", min: ArchiveConfig.TITLE_MIN)
-  validates_length_of :name,
-    maximum: ArchiveConfig.TITLE_MAX,
-    too_long: ts("must be less than %{max} characters long.", max: ArchiveConfig.TITLE_MAX)
-  validates_format_of :name,
-    message: ts('must begin and end with a letter or number; it may also contain underscores. It may not contain any other characters, including spaces.'),
-    with: /\A[A-Za-z0-9]\w*[A-Za-z0-9]\Z/
-  validates_length_of :icon_alt_text, allow_blank: true, maximum: ArchiveConfig.ICON_ALT_MAX,
-    too_long: ts("must be less than %{max} characters long.", max: ArchiveConfig.ICON_ALT_MAX)
-  validates_length_of :icon_comment_text, allow_blank: true, maximum: ArchiveConfig.ICON_COMMENT_MAX,
-    too_long: ts("must be less than %{max} characters long.", max: ArchiveConfig.ICON_COMMENT_MAX)
+  validates :name, presence: { message: I18n.t("collection.validation.name.presence") }
+  validates :name, uniqueness: { message: I18n.t("collection.validation.name.uniqueness") }
```

Comment:
I can add write a migration script to add a database constraint so rubocop stops complaining about this, but it feels kinda weird to do that just to please the linter... should I do that or suppress the warning in some way?

Reply:
Let's not, that would go completely out of the scope of the original PR. Furthermore, please revert 450dbeb5b9d068841d91fc5f143b9234a22390bf and ec338b0cf2b7c00da62ceec451504a15736aa105. While the dedication to i18n is appreciated, it would take a lot of extra work to [correctly localise](https://github.com/otwcode/otwarchive/wiki/Internationalization-%28i18n%29-Standards#user-content-Models) the errors and this PR is already quite large. We also risk a bug slipping through because manual QA wont test this i18n.

Reviewdog/Rubocop will complain about the ts, but that's fine since we're aware that we wont be handling those in this PR.

Reply:
Ok, I've reverted the requested commits